org.mixedrealitytoolkit.uxcore/Button/PressableButton.cs

```diff
@@ -142,12 +145,18 @@ public SpaceMode DistanceSpaceMode
         public bool RejectZRollOff { get => rejectZRollOff; set => rejectZRollOff = value; }
 
         /// <summary>
-        ///  Speed for extending the moving button visuals when selected by a non-touch source.
+        /// Speed for extending the moving button visuals when selected by a non-touch source.
         /// </summary>
         [SerializeField]
         [Tooltip("Speed for extending the moving button visuals when selected by a non-touch source.")]
         private float extendSpeed = 0.5f;
 
+        /// <summary>
+        /// Array with Components (set in Editor) that will be enabled/disabled based on proximity.
+        /// </summary>
+        [field: SerializeField, Tooltip("Array with Components (set in Editor) that will be enabled/disabled based on proximity.")]
+        public Component[] ProximityEnabledComponents { get; set; } = null;
```

Comment:
Maintainers (@keveleigh , @david-c-kline , @whebertML , @ghazen-ml , @shaynie ), I'd like your opinion on this.

I more in favor of adding `InProximityStarted`  and `InProximityStopped` events, and adding event handlers to hide/show button components.   Such events would fall more inline with the existing `StatefulInteractable` events.  

Also, different prefabs may want to do something other than enable/disable components on proximity.  Such events would make that possible.

Reply:
We could have both :) :   Add event handlers for InProximityStarted and InProximityStopped events so the customers can create custom behaviors and use the Component[] ProximityEnabledComponents array for the default enable/disable on proximity behavior.

Reply:
@ms-RistoRK @AMollis 
Clarification, are we intending to also turn off the whole parent plate background (the blue background for all elements as seen in gif below).  I initially thought this was only to enable/disable the proximity highlight and rounded outline of each element that reacts to proximity - not the parent rounded plate background as well.
![MRTK3_DynamicCanvas1](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/58860572/74b79f62-7958-4aa2-92c6-c7ebeb043502)

Reply:
> @ms-RistoRK @AMollis Clarification, are we intending to also turn off the whole parent panel background (the blue background for all elements as seen in gif below). I initially thought this was only to enable/disable the proximity highlight and rounded outline of each element that reacts to proximity - not the parent rounded panel background as well. ![MRTK3_DynamicCanvas1](https://private-user-images.githubusercontent.com/58860572/301676148-74b79f62-7958-4aa2-92c6-c7ebeb043502.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4MzQ3MDcsIm5iZiI6MTcwNjgzNDQwNywicGF0aCI6Ii81ODg2MDU3Mi8zMDE2NzYxNDgtNzRiNzlmNjItNzk1OC00YWEyLTkyYzYtYzdlYmViMDQzNTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDAwNDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMTAwODU2ZTY3MzViOTNkMzZmNGU2YmUxM2VjZDVmMzllYmVjMTBmZGZjYzkwZTBhM2U5ODEwZmU5NTYxOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jp4IVFd-mQaKDw-De2YLH49NE8TQA4Dc9zPwK6i2Jhs) [ ![MRTK3_DynamicCanvas1](https://private-user-images.githubusercontent.com/58860572/301676148-74b79f62-7958-4aa2-92c6-c7ebeb043502.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4MzQ3MDcsIm5iZiI6MTcwNjgzNDQwNywicGF0aCI6Ii81ODg2MDU3Mi8zMDE2NzYxNDgtNzRiNzlmNjItNzk1OC00YWEyLTkyYzYtYzdlYmViMDQzNTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDAwNDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMTAwODU2ZTY3MzViOTNkMzZmNGU2YmUxM2VjZDVmMzllYmVjMTBmZGZjYzkwZTBhM2U5ODEwZmU5NTYxOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jp4IVFd-mQaKDw-De2YLH49NE8TQA4Dc9zPwK6i2Jhs) ](https://private-user-images.githubusercontent.com/58860572/301676148-74b79f62-7958-4aa2-92c6-c7ebeb043502.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4MzQ3MDcsIm5iZiI6MTcwNjgzNDQwNywicGF0aCI6Ii81ODg2MDU3Mi8zMDE2NzYxNDgtNzRiNzlmNjItNzk1OC00YWEyLTkyYzYtYzdlYmViMDQzNTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDAwNDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMTAwODU2ZTY3MzViOTNkMzZmNGU2YmUxM2VjZDVmMzllYmVjMTBmZGZjYzkwZTBhM2U5ODEwZmU5NTYxOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jp4IVFd-mQaKDw-De2YLH49NE8TQA4Dc9zPwK6i2Jhs) [ ](https://private-user-images.githubusercontent.com/58860572/301676148-74b79f62-7958-4aa2-92c6-c7ebeb043502.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4MzQ3MDcsIm5iZiI6MTcwNjgzNDQwNywicGF0aCI6Ii81ODg2MDU3Mi8zMDE2NzYxNDgtNzRiNzlmNjItNzk1OC00YWEyLTkyYzYtYzdlYmViMDQzNTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDAwNDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMTAwODU2ZTY3MzViOTNkMzZmNGU2YmUxM2VjZDVmMzllYmVjMTBmZGZjYzkwZTBhM2U5ODEwZmU5NTYxOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jp4IVFd-mQaKDw-De2YLH49NE8TQA4Dc9zPwK6i2Jhs)

@whebertML Thank you for looking into this.  The requirements have changed from the original PR.  Making the canvas' CanvasRoundedRect component dynamic is not a requirement anymore.  Actually, the new approach is that the user selects which component(s) to make dynamic on proximity.
Note 1:  No button prefabs have any components set as dynamic btw.  The original behavior is kept as before this PR.
Note 2:  A new requirement to make components dynamic based on OnHover* and OnFocus* events just came in today which I will implement in the following days.
Note 3:  I updated CanvasExampleWithDynamicComponents.unity scene so that only the button's FrontPlate RawImage component is dynamic, the canvas' CanvasRoundedRect is no longer set as dynamic (also as per new requirement).

Reply:
@ms-RistoRK   Can you possibly share the new requirements?  Sounds like it will be using an event-based system like what @AMollis suggested above?  My personal opinion is that that would be more desirable than making PressableButton implement something like INearInteractionModeTarget.

It seems more ideal to provide such generic events for an Interactable that the dev could use to do whatever, including making certain visual components enabled/disabled like in this case - and so something like PressableButton need not even know about it.

Reply:
> @ms-RistoRK Can you possibly share the new requirements? Sounds like it will be using an event-based system like what @AMollis suggested above? My personal opinion is that that would be more desirable than making PressableButton implement something like INearInteractionModeTarget.
> 
> It seems more ideal to provide such generic events for an Interactable that the dev could use to do whatever, including making certain visual components enabled/disabled like in this case - and so something like PressableButton need not even know about it.

I'll make the change to an event-based approach.  As per the requirements, the original requirements were:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/088d1b45-6e14-4f3c-8331-15a9d16055eb)
This have evolved in time considering benefit for customers, flexibility, and expanding capabilities.  I have tried to update the PR main comment at the very top to reflect the latest decisions on our discussions.  The internal Tasks and User Story that is tracking all these changes and rescoping I cannot change because AFAIK they are classified as Confidential Microsoft information.  However, I'll keep updating the PR's main comment at the top to reflect the latest proposals.

Reply:
I have updated the code to use an event-based approach and disable dynamic components iff there is no hovering nor proximity occurring.  Please re-review PR.